### ISSUE TITLE
Pinpoint notifications not opening app targeting Android 12

### ISSUE THREAD

**Issue by cirion** [aws-amplify/aws-sdk-android#2762]
**Describe the bug**
When targeting Android SDK 31, notifications appear successfully, but tapping them just dismisses the notification without opening the app.

**To Reproduce**

Set up push notifications as described in the [docs](https://docs.amplify.aws/sdk/push-notifications/messaging-campaign/q/platform/android/).

Set the app `targetSdk 31`.

Send a push notification to the device, configured to open the app or a URL.

Tap on the push notification.

**Which AWS service(s) are affected?**

Pinpoint notifications for Android.

**Expected behavior**
Tapping the notification should open the app, deeplink, or URL set in the notification.

**Environment Information (please complete the following information):**
 - AWS Android SDK Version: 2.37.1
 - Device: Various, including Pixel 6
 - Android Version: 12 / API 31
 - Specific to simulators: No

**Additional context**

Opening the app from a notification works fine with the targetSdk set to 30.

The relevant part of the log seems to be the following:

```
I/ActivityTaskManager: START u0 {act=android.intent.action.VIEW dat=https://example.com/... flg=0x10000000 cmp=org.mozilla.firefox/org.mozilla.fenix.IntentReceiverActivity} from uid 10335
E/NotificationService: Indirect notification activity start (trampoline) from com.example.android.debug blocked
W/ActivityTaskManager: Background activity start [callingPackage: com.example.android.debug; callingUid: 10335; appSwitchAllowed: true; isCallingUidForeground: false; callingUidHasAnyVisibleWindow: false; callingUidProcState: RECEIVER; isCallingUidPersistentSystemProcess: false; realCallingUid: 10335; isRealCallingUidForeground: false; realCallingUidHasAnyVisibleWindow: false; realCallingUidProcState: RECEIVER; isRealCallingUidPersistentSystemProcess: false; originatingPendingIntent: null; allowBackgroundActivityStart: false; intent: Intent { act=android.intent.action.VIEW dat=https://example.com/... flg=0x10000000 cmp=org.mozilla.firefox/org.mozilla.fenix.IntentReceiverActivity }; callerApp: ProcessRecord{efeb5fb 28572:com.example.android.debug/u0a335}; inVisibleTask: false]
E/ActivityTaskManager: Abort background activity starts from 10335
```

This seems to be related to a [change in Android 12](https://developer.android.com/about/versions/12/behavior-changes-12#notification-trampolines) to disallow starting activities from services or receivers that were opened by a notification.

**Comment by tjleing** [comment]
I was able to reproduce this issue.  Looks like @sktimalsina updated the `PendingIntent` code on showing the notification, but actually processing opening the notification was not updated yet.  As mentioned in #2326 a fix for this might end up fixing both.

**Comment by sokarcreative** [comment]
Any news ? I have the same problem too.

**Comment by mattlenehan** [comment]
🙋‍♂️ I'm having this problem too, please help 🙏

**Comment by nzayatz14** [comment]
I also am having this issue

**Comment by tylerjroach** [comment]
This issue has been resolved and is live in v2.48.0.
